Add resource collection callout behavior (#31193)
DataDog/documentation: content/en/datadog_cloudcraft/_index.md

**Note**: Cloudcraft adapts to restrictive permissions by excluding inaccessible resources. For example, if you don't grant permission to list S3 buckets, the diagram excludes those buckets. If permissions block certain resources, an alert displays in the UI.

> <div class="alert alert-info"><strong>Note</strong>: Enabling resource collection for your AWS accounts may impact your AWS CloudWatch costs if you are using Usage Metrics. To avoid the additional charge, you can disable collecting "Usage" metrics in the Metric Collection tab of the <a href="https://app.datadoghq.com/integrations/amazon-web-services">Datadog AWS Integration.</a><br/>

Reviewer: I would consider elevating this alert to a warning (yellow) since it impacts costs.

> <div class="alert alert-info"><strong>Note</strong>: Enabling resource collection for your AWS accounts may impact your AWS CloudWatch costs if you are using Usage Metrics. To avoid the additional charge, you can disable collecting "Usage" metrics in the Metric Collection tab of the <a href="https://app.datadoghq.com/integrations/amazon-web-services">Datadog AWS Integration.</a><br/>

Reviewer: ```suggestion
<div class="alert alert-info"><strong>Note</strong>: Enabling resource collection can impact your AWS CloudWatch costs. To avoid these charges, disable <strong>Usage</strong> metrics in the <strong>Metric Collection</strong> tab of the <a href="https://app.datadoghq.com/integrations/amazon-web-services">Datadog AWS Integration</a>.<br/>
```
^ Just an idea for an edit as well. Otherwise, I'd recommend lowercasing "Usage Metrics", removing quotes around "Usage", and bolding the tab name.

> {{< img src="/infrastructure/resource_catalog/aws_usage_toggle.png" alt="AWS Usage toggle in account settings" style="width:100%;" >}}</div>

Reviewer: This is a pretty big screenshot, especially for an alert box. You might consider cropping it to just the Metric Collection tab content.
